### PR TITLE
✨ [extension] revamp event description and JSON viz

### PR DESCRIPTION
**Pull request by BenoitZugmeyer**
## Motivation

The current JSON rendering component that we use does not allow to display a proper event description. Also it is a bit limited for future extensions.

## Changes

Implement a new Json component and a better event description similar to the Datadog app

### Before

Screenshot 2023-08-24 at 12 23 11

### After

Screenshot 2023-08-24 at 12 21 50

## Testing

- [x] Local
- [ ] Staging
- [ ] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

**Comment by BenoitZugmeyer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2369** <a href="https://app.graphite.dev/github/pr/DataDog/browser-sdk/2369" target="_blank">Graphite</a> 
    * **PR #2370** <a href="https://app.graphite.dev/github/pr/DataDog/browser-sdk/2370" target="_blank">Graphite</a> 
      * **PR #2371** <a href="https://app.graphite.dev/github/pr/DataDog/browser-sdk/2371" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DataDog/browser-sdk/2371?utm_source=stack-comment).

File: developer-extension/src/panel/components/json.tsx [hunk 1]
```diff
@@ -1,58 +1,375 @@
...
+  ) => {
+    const theme = useMantineTheme()
+    return (
+      <Box
+        ref={
+          // setting a HTMLDivElement | HTMLSpanElement ref messes with TS types as the component prop is also div | span
+          ref as any
+        }
+        {...boxProps}
+        component={doesValueHasChildren(value) ? 'div' : 'span'}
```
Comment (amortemousque) [inline developer-extension/src/panel/components/json.tsx:79]: **❓ question:** ‏Why is this necessary?

File: developer-extension/src/panel/components/tabs/eventsTab/eventRow.tsx [hunk 2]
```diff
@@ -21,9 +32,33 @@ const LOG_STATUS_COLOR = {
...
-    <tr>
+    <tr
+      onClick={(event) => {
+        if (jsonRef.current?.contains(event.target as Node)) {
+          // Ignore clicks on the collapsible area
+          return
+        }
+        setIsCollapsed((previous) => !previous)
+      }}
+    >
```
Comment (bcaudan) [inline developer-extension/src/panel/components/tabs/eventsTab/eventRow.tsx:61]: **💬 suggestion:** ‏Could we add back the pointer cursor on the event row?
users could easily miss that it is interactive otherwise

File: developer-extension/src/panel/formatNumber.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export function formatNumber(n: number): string {
```
Comment (bcaudan) [inline developer-extension/src/panel/formatNumber.ts:1]: **💬 suggestion:** `1,692,964,306,392` display for timestamps seems odd to me.
what about having a specific format for those?

File: developer-extension/src/panel/components/json.tsx (hunk 1 above)
Comment (BenoitZugmeyer) [reply developer-extension/src/panel/components/json.tsx:79]: The value is displayed inline if it does not have children, ex:

Screenshot 2023-09-07 at 11 36 23

WDYT? Is it overkill?

**Comment by BenoitZugmeyer** [comment]
Current dependencies on/for this PR:

* main:
  * **PR #2369**
    * **PR #2370**
      * **PR #2371** 👈
        * **PR #2372**

This comment was autogenerated by Freephite.

File: developer-extension/src/panel/components/tabs/eventsTab/eventRow.tsx (hunk 2 above)
Comment (BenoitZugmeyer) [reply developer-extension/src/panel/components/tabs/eventsTab/eventRow.tsx]: It should be fixed

**Review by bcaudan — APPROVED** [review]

**Review by amortemousque — APPROVED** [review]
